globus_cli/parsing/shared_options.py

```diff
@@ -244,6 +244,33 @@ def validate_endpoint_create_and_update_params(endpoint_type, managed, params):
                     ("Option --{} can only be used with managed "
                      "endpoints".format(option.replace("_", "-"))))
 
+    # because the Transfer service doesn't do network use level updates in a
+    # patchy way, *both* endpoint `POST`s *and* `PUT`s must either use
+    # - `network_use='custom'` with *every* other parameter specified, or
+    # - a preset/absent `network_use` with *no* other parameter specified (in
+    #   this case, Transfer will accept but ignore the others parameters if
+    #   given, leading to user confusion if we don't do this validation check)
+    custom_network_use_params = ('max_concurrency', 'preferred_concurrency',
+                                 'max_parallelism', 'preferred_parallelism')
+    if params['network_use'] == 'custom':
+        for option in custom_network_use_params:
```

Comment:
This part of the "if" is already checked by transfer-api.  It will fail if custom doesn't pass all parameters.  Probably not harmful if you check it yourself, but hosted CLI does not.

Reply:
Given our recent re-learned lessons about duplicating validation client-side and service-side, I am inclined to agree. Unless @sirosen wants me to keep this for some reason, I will drop this part and just run the check below `if params['network_use'] != 'custom'`.

Reply:
If the API rejects this usage, then yes, I agree with both of you that we should let the command through and allow for the API error.